src/tests/multihost/alltests/test_krb_access_provider.py

```diff
@@ -0,0 +1,258 @@
+    domain_section = f'domain/{ds_instance_name}'
+    domain_params = {'access_provider': 'krb5',
+                     'use_fully_qualified_names': 'False',
+                     'override_homedir': '/home/%u'}
+    tools.sssd_conf(domain_section, domain_params)
+    tools.clear_sssd_cache()
+
+    sed_cmd = f'sed -i "s/kerberos.example.com/{session_multihost.master[0].sys_hostname}/" /etc/krb5.conf'
+    session_multihost.client[0].run_command(sed_cmd)
+    sed_cmd = 'sed -i "s:^#::g" /etc/krb5.conf'
+    session_multihost.client[0].run_command(sed_cmd)
+
+    restore_selinux = "restorecon -v /etc/krb5.keytab"
+    session_multihost.client[0].run_command(restore_selinux)
+
+    cmds = ["mkdir -p /home/foo3",
+            "chown foo3: /home/foo3",
+            "chmod 700 /home/foo3",
+            "truncate -s 0 /home/foo3/.k5login",
+            "chown foo3: /home/foo3/.k5login",
+            "restorecon -v /home/foo3/.k5login"]
+    for command in cmds:
+        cmd = session_multihost.client[0].run_command(command)
+        assert cmd.returncode == 0
+
+
+@pytest.mark.krbaccessprovider
+@pytest.mark.usefixtures('custom_setup')
+class TestKrbAccessProvider(object):
+    """
+    This is test case class for krb_access_provider suite
+
+    Test access provision by kerberos on the basis of .k5login file.
+    Test for empty k5login file and test for presence of single or 
+    multiple kerberos principal in the file, test for deleted user with
+    principals present in the file.
+    """
+    @staticmethod
+    def test_0001_k5login_empty(multihost):
+        """
+        :title: Create an empty .k5login file in the user's home directory
+        :id: e84bd024-669f-40d0-8833-edbc1be48e8e
+        :setup:
+          1. Clear the secure log, clear cache and restart sssd.
+        :steps:
+          1. Authenticate the ldapusers from the client
+          2. Check the secure log for expected messages.
+          3. Check the sssd domain log for expected messages.
+        :expectedresults:
+          1. Secure Log contains the expected lines:
+             pam_sss(sshd:auth): authentication success for foo3
+             pam_sss(sshd:account): Access denied for user foo3
+
+             pam_sss(sshd:auth): authentication success for foo4
+             Accepted password for foo4
+          2. SSSD Domain Log contains the expected lines:
+             Access denied for user [foo3
+             Access allowed for user [foo4
+        :teardown:
+          1. Clear the .k5login file.
+        """
+        multihost.client[0].run_command("truncate -s 0 /var/log/secure")
+        client = sssdTools(multihost.client[0])
+        client.clear_sssd_cache()
+
+        users = ['foo3', 'foo4']
+        ssh = client.auth_from_client(users[0], 'Secret123')
+        assert ssh == 10, "foo3 successfully logged In"
+        ssh = client.auth_from_client(users[1], 'Secret123')
+        assert ssh == 3, "foo4 failed to log In"
+
+        time.sleep(15)
```

Comment:
We should try to avoid sleeps whenever possible, I understand it was in the old bash test, but is it really needed here? Can you try without it, or with a lower value like `5` ?

Reply:
I can try with lower value but since before the logs getting updated authentication is happening so thought to provide a reasonable amount of time.